core/src/main/java/software/amazon/awssdk/auth/ProfileCredentialsProvider.java

```diff
@@ -70,7 +71,12 @@ public static Builder builder() {
 
     @Override
     protected AwsCredentials loadCredentials() {
-        return profilesConfigFile == null ? null : profilesConfigFile.getCredentials(profileName);
+        if (profilesConfigFile == null) {
+            throw new SdkClientException("Unable to find profile configuration file. If you are specifying a profile config "
+                                         + "file please verify it exists.");
+        }
+
+        return profilesConfigFile.getCredentials(profileName);
```

Comment:
Do we throw a nice error message if the profile name is bad?

Reply:
"No AWS profile named {name}"